### PR TITLE
Adds Doxygen annotations to epicsType.h

### PR DESCRIPTION
**Pull request by DominicOram**
As per the codeathon project detailed in https://github.com/epics-base/epics-base/wiki/Header-annotations-project.

I modified the pre-existing comments and added a few more where it was obvious to do so.

### CODE REVIEW

File: modules/libcom/src/misc/epicsTypes.h [hunk 1]
```diff
@@ -49,15 +54,15 @@ typedef epicsUInt16     epicsEnum16;
...
 
 typedef struct {
     unsigned    length;
     char        *pString;
 } epicsString;
 
-/*
- * !! Dont use this - it may vanish in the future !!
+/**
+ * \brief !! Dont use this - it may vanish in the future !!
```
Comment (mdavidsaver) [inline modules/libcom/src/misc/epicsTypes.h:70]: @anjohnson Is this warning still appropriate?  Use of `MAX_STRING_SIZE` remains common.

Comment (anjohnson) [reply modules/libcom/src/misc/epicsTypes.h:70]: I would like to see a "Don't use this" comment on the `epicsOldString` and `epicsString` typedefs, but not on the `MAX_STRING_SIZE` macro which is far too widely used to be deprecated.

Comment (mdavidsaver) [reply modules/libcom/src/misc/epicsTypes.h:70]: Agreed.

Comment (DominicOram) [reply modules/libcom/src/misc/epicsTypes.h:70]: I'm happy to change this tomorrow

**Review by mdavidsaver — APPROVED** [review]
Looks ok to me.

**Comment by mdavidsaver** [comment]
Applied as 1fcbdad5e991af5b8c7a4f6996c9a301e849bfc1.  Thanks.

**Pull request closed without merging**